src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,22 +1,34 @@
-import {Component} from '@angular/core';
-import {productsMock} from '../../../shared/products/products.mock';
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import {IProduct} from '../../../shared/products/product.interface';
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
 })
-export class CardComponent {
-    readonly product = productsMock[0];
+export class CardComponent implements OnInit {
+    @Input({required: true})
+    product: IProduct | null = null;
+
+    @Output()
+    productId = new EventEmitter<string>();
+
+    imageUrl = '';
+
+    ngOnInit(): void {
+        if (this.product?.images) {
+            this.imageUrl = this.product.images[0].url;
+        }
+    }
 
     onProductBuy(event: Event) {
         event.stopPropagation();
 
         // eslint-disable-next-line no-console
-        console.log('Buy product');
+        this.productId.emit(this.product?._id);
     }
 
-    isStarActive(starIndex: number): boolean {
-        return this.product.rating >= starIndex;
+    isStarActive(starIndex: number): boolean | null {
+        return this.product ? this.product?.rating >= starIndex : null;
```

Comment:
Лучше из данной функции возвращать только `boolean`, так она проще воспринимается
```suggestion
    isStarActive(starIndex: number): boolean {
        return this.product ? this.product?.rating >= starIndex : false;
```

Reply:
исправила